### PR TITLE
Remove CORS headers from pod proxy responses

### PR DESCRIPTION
**Pull request by csrwng**
The API server sends its own CORS headers in its response, and if the proxied pod response also includes its own headers, clients get confused.

### CODE REVIEW

**Comment by csrwng** [comment]
@lavalamp ptal
@liggitt fyi -- addresses https://github.com/openshift/origin/issues/2619

File: pkg/registry/generic/rest/proxy.go [hunk 1]
```diff
@@ -214,9 +214,37 @@ func (h *UpgradeAwareProxyHandler) defaultProxyTransport(url *url.URL) http.Roun
...
 		Scheme:      scheme,
 		Host:        host,
 		PathPrepend: pathPrepend,
 	}
+	return &upgradeAwareProxyTransport{
+		RoundTripper: internalTransport,
+	}
+}
+
+// upgradeAWareProxyTransport is a wrapper for an internal transport. It removes CORS headers
```
Comment (lavalamp) [inline pkg/registry/generic/rest/proxy.go]: Perhaps 'corsRemovingTransport'?

File: pkg/registry/generic/rest/proxy_test.go [hunk 2]
```diff
@@ -71,9 +73,11 @@ func validateParameters(t *testing.T, name string, actual url.Values, expected m
 	}
 }
 
-func validateHeaders(t *testing.T, name string, actual http.Header, expected map[string]string) {
+func validateHeaders(t *testing.T, name string, actual http.Header, expected map[string]string, notExpected []string) {
+	checked := map[string]bool{}
```
Comment (lavalamp) [inline pkg/registry/generic/rest/proxy_test.go]: unneeded variable?

File: pkg/registry/generic/rest/proxy_test.go [hunk 3]
```diff
@@ -135,7 +167,7 @@ func TestServeHTTP(t *testing.T) {
 			backendResponse := "<html><head></head><body><a href=\"/test/path\">Hello</a></body></html>"
 			backendHandler := &SimpleBackendHandler{
 				responseBody:   backendResponse,
-				responseHeader: map[string]string{"Content-Type": "text/html"},
+				responseHeader: test.responseHeader,
```
Comment (lavalamp) [inline pkg/registry/generic/rest/proxy_test.go]: Either default it to also set (and check) "Content-Type": "text/html", or add that to every entry in the table... :/

Can't stop testing that behavior.

File: pkg/registry/generic/rest/proxy_test.go [hunk 4]
```diff
@@ -209,6 +245,8 @@ func TestServeHTTP(t *testing.T) {
 				t.Errorf("Did not get expected response body: %s. Expected: %s", rb, backendResponse)
 			}
 
+			// Validate response headers
```
Comment (lavalamp) [inline pkg/registry/generic/rest/proxy_test.go]: stray comment?

**Comment by csrwng** [comment]
@lavalamp thank you for taking a look at it. I've addressed your comments.

**Comment by lavalamp** [comment]
ok to test

**Comment by lavalamp** [comment]
LGTM

**Comment by rhuss** [comment]
The fix doesn't seem to work anymore, please see https://github.com/openshift/origin/issues/2619 for details.

**Pull request merged**